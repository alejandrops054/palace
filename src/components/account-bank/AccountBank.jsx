/* Sistema Intelligence 2.0
Módulo Contratos
Versión 1.0
Author: IntregraIT Contratos
Comentarios: La tabla esta incompleta revisar 18/01/2021
Agregando comnentario de testing
*/
import React, { useState } from "react";
import "antd/dist/antd.css";
import { Table } from "antd";
import {
  Col,
  Row,
  Card,
  CardBody,
  CardTitle,
  Button,
  ButtonGroup
} from "reactstrap";

export default function AccountBank() {
  const columns = [
    {
      title: "Code ",
      dataIndex: "code",
      key: "code",
      width: 100,
    },

    {
      title: "Description",
      dataIndex: "description",
      key: "description",
      width: 100,
    },
    {
      title: "Merchar Code",
      dataIndex: "merchar",
      key: "merchar",
      width: 100,
    },
    {
      title: "Account",
      dataIndex: "account",
      key: "account",
      width: 100,
    },
    {
      title: "Currency",
      dataIndex: "currency",
      key: "currency",
      width: 100,
    },
    {
      title: "Status",
      dataIndex: "status",
      key: "status",
      width: 100,
    },
    {
      title: "Account Location",
      dataIndex: "accountL",
      key: "accountL",
      width: 100,
    },
  ];
  const data = [
    {
      code: "1",
      description: "Santander",
      merchar: "7271271",
      account: "3121211",
      currency: "MX",
      status: "A",
      accountL: "",
    },
    {
        code: "1",
        description: "Santander",
        merchar: "7271271",
        account: "3121211",
        currency: "MX",
        status: "A",
        accountL: "",
      },
      {
        code: "1",
        description: "Santander",
        merchar: "7271271",
        account: "3121211",
        currency: "MX",
        status: "A",
        accountL: "",
      },
      {
        code: "1",
        description: "Santander",
        merchar: "7271271",
        account: "3121211",
        currency: "MX",
        status: "A",
        accountL: "",
      },
      {
        code: "1",
        description: "Santander",
        merchar: "7271271",
        account: "3121211",
        currency: "MX",
        status: "A",
        accountL: "",
      },
  ];
  return (
    <>
      <Card>
        <CardBody>
          <CardTitle>
            <h3>Account Bank</h3>
          </CardTitle>
          <br />
          <Row>
            <Col xs="8">
              <Table columns={columns} dataSource={data} />
            </Col>
          </Row>
          <Row>
          <Col xs="2">
              <span>Records: 8</span>
            </Col>
          <Col xs="7">
            </Col>
            <Col xs="3">
              <ButtonGroup>
                <Button color="primary">Log</Button>
                <Button color="primary">Add</Button>
                <Button color="primary">Properties</Button>
                <Button color="primary">Refresh</Button>
                <Button color="primary">Close</Button>
              </ButtonGroup>
            </Col>
          </Row>
        </CardBody>
      </Card>
    </>
  );
}
