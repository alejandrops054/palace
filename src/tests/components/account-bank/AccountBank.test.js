/* Sistema Intelligence 2.0
Módulo Contratos
Versión 1.0
Author: IntregraIT Contratos
Comentarios: xxxxx
*/
import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import AccountBank from '../../../components/account-bank/AccountBank'

describe('Pruebas del componente AccountBank ', () => {
    test('Creacion de AccountBank', () => {
        
        const wrapper = shallow(<AccountBank/>)

        expect(wrapper).toMatchSnapshot();
    })
    
})