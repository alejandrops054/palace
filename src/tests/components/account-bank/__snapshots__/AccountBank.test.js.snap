// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente AccountBank  Creacion de AccountBank 1`] = `
<Fragment>
  <Card
    tag="div"
  >
    <CardBody
      tag="div"
    >
      <CardTitle
        tag="div"
      >
        <h3>
          Account Bank
        </h3>
      </CardTitle>
      <br />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="8"
        >
          <Table
            columns={
              Array [
                Object {
                  "dataIndex": "code",
                  "key": "code",
                  "title": "Code ",
                  "width": 100,
                },
                Object {
                  "dataIndex": "description",
                  "key": "description",
                  "title": "Description",
                  "width": 100,
                },
                Object {
                  "dataIndex": "merchar",
                  "key": "merchar",
                  "title": "Merchar Code",
                  "width": 100,
                },
                Object {
                  "dataIndex": "account",
                  "key": "account",
                  "title": "Account",
                  "width": 100,
                },
                Object {
                  "dataIndex": "currency",
                  "key": "currency",
                  "title": "Currency",
                  "width": 100,
                },
                Object {
                  "dataIndex": "status",
                  "key": "status",
                  "title": "Status",
                  "width": 100,
                },
                Object {
                  "dataIndex": "accountL",
                  "key": "accountL",
                  "title": "Account Location",
                  "width": 100,
                },
              ]
            }
            dataSource={
              Array [
                Object {
                  "account": "3121211",
                  "accountL": "",
                  "code": "1",
                  "currency": "MX",
                  "description": "Santander",
                  "merchar": "7271271",
                  "status": "A",
                },
                Object {
                  "account": "3121211",
                  "accountL": "",
                  "code": "1",
                  "currency": "MX",
                  "description": "Santander",
                  "merchar": "7271271",
                  "status": "A",
                },
                Object {
                  "account": "3121211",
                  "accountL": "",
                  "code": "1",
                  "currency": "MX",
                  "description": "Santander",
                  "merchar": "7271271",
                  "status": "A",
                },
                Object {
                  "account": "3121211",
                  "accountL": "",
                  "code": "1",
                  "currency": "MX",
                  "description": "Santander",
                  "merchar": "7271271",
                  "status": "A",
                },
                Object {
                  "account": "3121211",
                  "accountL": "",
                  "code": "1",
                  "currency": "MX",
                  "description": "Santander",
                  "merchar": "7271271",
                  "status": "A",
                },
              ]
            }
            rowKey="key"
          />
        </Col>
      </Row>
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="2"
        >
          <span>
            Records: 8
          </span>
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="7"
        />
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="3"
        >
          <ButtonGroup
            role="group"
            tag="div"
          >
            <Button
              color="primary"
              tag="button"
            >
              Log
            </Button>
            <Button
              color="primary"
              tag="button"
            >
              Add
            </Button>
            <Button
              color="primary"
              tag="button"
            >
              Properties
            </Button>
            <Button
              color="primary"
              tag="button"
            >
              Refresh
            </Button>
            <Button
              color="primary"
              tag="button"
            >
              Close
            </Button>
          </ButtonGroup>
        </Col>
      </Row>
    </CardBody>
  </Card>
</Fragment>
`;
